umh-core/pkg/fsm/redpanda/actions.go

```diff
@@ -321,7 +334,12 @@ func (r *RedpandaInstance) UpdateObservedStateOfInstance(ctx context.Context, se
 			r.baseFSMInstance.GetLogger().Debugf("Configuration differences: %s", diffStr)
 
 			// Update the config in the S6 manager
-			err := r.service.UpdateRedpandaInS6Manager(ctx, &r.config, r.baseFSMInstance.GetID())
+			var needsRestart bool
+			needsRestart, err = r.service.UpdateRedpandaInS6Manager(ctx, &r.ObservedState.ObservedRedpandaServiceConfig, &r.config, r.baseFSMInstance.GetID(), tick)
+			if needsRestart {
+				r.baseFSMInstance.GetLogger().Debugf("Redpanda service %s needs to be restarted", r.baseFSMInstance.GetID())
+				return r.RestartInstance(ctx, services.GetFileSystem())
```

Comment:
this here could then send a "restart" event, which then causes it to be moved into to_be_restarted. after that it goes into restarting, and as soon as it is confirmed that the restart worked, it should go into starting state (so we can leverage the safety checks there)

Reply:
Not sure if we even need that.
Since RestartInstance now sends the Stop event, S6 will shut it down, and once it is in the shutdown state, we will automatically reconcile to active again.